utils/src/main/java/software/amazon/awssdk/utils/cache/NonBlocking.java

```diff
@@ -61,23 +67,27 @@ public class NonBlocking implements CachedSupplier.PrefetchStrategy {
      * the {@link #EXECUTOR}.
      */
     private static final ScheduledThreadPoolExecutor SCHEDULER =
-        new ScheduledThreadPoolExecutor(1, new ThreadFactoryBuilder().daemonThreads(true).build());
+        new ScheduledThreadPoolExecutor(1, new ThreadFactoryBuilder().threadNamePrefix("sdk-cache-scheduler")
+                                                                     .daemonThreads(true)
+                                                                     .build());
 
     /**
      * Threads used to do the actual work of refreshing the values (because the cached supplier might block, so we don't
      * want the work to be done by a small thread pool). This executor is created as unbounded, but we start complaining and
      * skipping refreshes when there are more than {@link #MAX_CONCURRENT_REFRESHES} running.
      */
     private static final ThreadPoolExecutor EXECUTOR =
-        new ThreadPoolExecutor(0, Integer.MAX_VALUE,
+        new ThreadPoolExecutor(1, Integer.MAX_VALUE,
                                60L, TimeUnit.SECONDS,
                                new SynchronousQueue<>(),
-                               new ThreadFactoryBuilder().daemonThreads(true).build());
+                               new ThreadFactoryBuilder().threadNamePrefix("sdk-cache")
```

Comment:
Perhaps a name of `aws-sdk-credentials-provider-cache-prefetch`

Reply:
The latest revision prepends this to the thread name specified when the prefetch strategy is created, so the complete thread name in logs would be something like sdk-cache-0-sts-assume-role-credentials-provider-0. , and idle threads would be sdk-cache-0.

Does that make this better?